pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java

```diff
@@ -134,12 +134,15 @@ public void run() throws Exception {
     Path outputDir = new Path(_outputDir);
 
     if (fs.exists(stagingDir)) {
-      LOGGER.warn("Found the temp folder, deleting it");
+      LOGGER.warn("Found the temp folder {}, deleting it", stagingDir);
       fs.delete(stagingDir, true);
     }
     fs.mkdirs(stagingDir);
-    if (_defaultPermissions != null) {
-      FileSystem.get(getConf()).setPermission(stagingDir, new FsPermission(_defaultPermissions));
+
+    if (_defaultPermissionsMask != null) {
+      FsPermission umask = new FsPermission(_defaultPermissionsMask);
```

Comment:
Creation of directory with specified can be extracted out as a method that can be unit tested that compares the umask value specified in the test with the permissions created for the directory.

Reply:
I exercised this manually in the integration test. Unit tests for this portion would include a lot of mocking because I need to get a hadoop filesystem, so I think the first way is preferable.

Reply:
It can be unit tested for PinotLocalFs, which should not require mocking?